meteor/server/api/playout/adlib.ts

```diff
@@ -22,28 +22,29 @@ import { rundownPlaylistSyncFunction, RundownSyncFunctionPriority } from '../ing
 
 import { PieceInstances, PieceInstance, PieceInstanceId } from '../../../lib/collections/PieceInstances'
 import { PartInstances, PartInstance, PartInstanceId } from '../../../lib/collections/PartInstances'
+import { initCacheForRundownPlaylist, CacheForRundownPlaylist } from '../../DatabaseCaches'
 
 export namespace ServerPlayoutAdLibAPI {
-	export function pieceTakeNow (rundownPlaylistId: RundownPlaylistId, partInstanceId: PartInstanceId, pieceInstanceIdOrPieceIdToCopy: PieceInstanceId | PieceId) {
-		return rundownPlaylistSyncFunction(rundownPlaylistId, RundownSyncFunctionPriority.USER_PLAYOUT, () => {
-			const rundownPlaylist = RundownPlaylists.findOne(rundownPlaylistId)
```

Comment:
Not doing a full review, just saw this while dealing with a merge conflict

I don't like the rundownPlaylist crossing the syncFunction boundary like this. It could cause all kinds of weird race conditions, as any other rundownPlaylistSyncFunction which are already executing or queued could change any of the properties set on the playlist.
Once inside the rundownPlaylistSyncFunction, it has the lock to the RundownPlaylist so can reload and be guaranteed that nothing else will change it (in any way that matters)

example timeline:
0.0s User does take to part 'b' (currently in 'a')
0.1s User runs adlib in part 'a'
0.2s Core starts take
0.4s Core queues 'run adlib' call (at this rundownPlaylistSyncFunction). 
5.0s Core completes take. current part is now 'b'
5.1s Core starts this 'run-adlib'.
	rundownPlaylist.currentPartInstance is now not in sync with the db, so does the wrong thing